### PR TITLE
Enemy

### LINKED ISSUE #21: Implement Enemy
Implement pac-man style enemy that locks onto and chases a player. Use a navmesh for path finding. Set the enemy speed to slower than players so they may evade.

Estimated time: 2-5 hours

### PR DESCRIPTION
**Pull request by Victor-Avelar**
closes #21 

Also wrote a devlog.

### CODE REVIEW

File: _posts/2020-05-19-pathfinding-devlog.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Turns out, navmesh agents aren't supported for 2D. After some minutes googling, it appears that there are a couple of 
+solutions available to me.
+
+1. Make the project 3D and pretend it's 2D, putting sprites on 3D objects. 
+Ok, cool, maybe next project, but we have done too much work in 2D at this point to convert everything.
+
+2. Import third party navmesh components for 2D.
+Open source is the best. Turns out other people have already solved this problem for me. There are a couple components out there,
+one we found in this thread: https://answers.unity.com/questions/1492606/unity-2d-navmesh-for-a-tilemap.html
```
Comment (etcadinfinitum) [inline _posts/2020-05-19-pathfinding-devlog.md:26]: Use markdown link embedding for super sweet embedded links

```suggestion
one we found [in this thread](https://answers.unity.com/questions/1492606/unity-2d-navmesh-for-a-tilemap.html)
```

File: _posts/2020-05-19-pathfinding-devlog.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: Pathfinding for Enemy chase behavior, Difficulty with 2D navmashes
+description: Containing instructions for writing up posts for the development log.
+---
+
+# Development Log Dated 05-23-2020
```
Comment (etcadinfinitum) [inline _posts/2020-05-19-pathfinding-devlog.md:7]: One nitpick -- don't post-date the date in the filename (per Pisan's instructions); I agree that it should be 5/23, just change the filename on this in another commit.

File: _posts/2020-05-19-pathfinding-devlog.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+1. Make the project 3D and pretend it's 2D, putting sprites on 3D objects. 
+Ok, cool, maybe next project, but we have done too much work in 2D at this point to convert everything.
+
+2. Import third party navmesh components for 2D.
+Open source is the best. Turns out other people have already solved this problem for me. There are a couple components out there,
+one we found in this thread: https://answers.unity.com/questions/1492606/unity-2d-navmesh-for-a-tilemap.html
+and the we decided to go with: https://github.com/h8man/NavMeshPlus
+
+NavMeshPlus by h8man is really well documented, and is ready to go by simply adding this dependency:
```
Comment (etcadinfinitum) [inline _posts/2020-05-19-pathfinding-devlog.md:29]: Add this dependency...where? In `ProjectSettings`?

File: _posts/2020-05-19-pathfinding-devlog.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+1. Make the project 3D and pretend it's 2D, putting sprites on 3D objects. 
+Ok, cool, maybe next project, but we have done too much work in 2D at this point to convert everything.
+
+2. Import third party navmesh components for 2D.
+Open source is the best. Turns out other people have already solved this problem for me. There are a couple components out there,
+one we found in this thread: https://answers.unity.com/questions/1492606/unity-2d-navmesh-for-a-tilemap.html
+and the we decided to go with: https://github.com/h8man/NavMeshPlus
+
+NavMeshPlus by h8man is really well documented, and is ready to go by simply adding this dependency: 
+"com.h8man.2d.navmeshplus": "https://github.com/h8man/NavMeshPlus.git#master".
```
Comment (etcadinfinitum) [inline _posts/2020-05-19-pathfinding-devlog.md:30]: Use triple backticks to format this as monospaced text. If it's a common language/syntax, you can reference it like so:

````
```unity
"com.h8man.2d.navmeshplus": "https://github.com/h8man/NavMeshPlus.git#master"
```
````

becomes:
```unity
"com.h8man.2d.navmeshplus": "https://github.com/h8man/NavMeshPlus.git#master"
```

````suggestion
```
"com.h8man.2d.navmeshplus": "https://github.com/h8man/NavMeshPlus.git#master"
```
````

File: _posts/2020-05-19-pathfinding-devlog.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: Pathfinding for Enemy chase behavior, Difficulty with 2D navmashes
+description: Containing instructions for writing up posts for the development log.
```
Comment (etcadinfinitum) [inline _posts/2020-05-19-pathfinding-devlog.md:4]:
```suggestion
title: Pathfinding and 2D Navmesh Challenges
description: We had to change our map philosophy to allow pathfinding for enemy NPCs; here's a short note about the work Victor did to get this figured out.
```

File: _posts/2020-05-19-pathfinding-devlog.md (hunk 3 above)
Comment (Victor-Avelar) [reply _posts/2020-05-19-pathfinding-devlog.md:29]: Package dependencies are located in trapped/Packages/manifest.json

File: _posts/2020-05-19-pathfinding-devlog.md (hunk 2 above)
Comment (Victor-Avelar) [reply _posts/2020-05-19-pathfinding-devlog.md]: I meant to write 23 but I was using your devlog as a template on how to format so my brain pulled a funny and wrote 19.

**Pull request merged**